@aggregate fails to compile on python 3.14

### Describe the bug

`@aggregate` does not work in python 3.14 due to the introduction of PEP 649. This is (in part) caused by the use of the `cls.__dict__.get("__annotations__", {})` idiom, which is specifically mentioned as may not working in https://peps.python.org/pep-0649/. This can be fixed by using `inspect.get_annotations()` instead as a drop-in replacement following https://docs.python.org/3/howto/annotations.html. The python docs best practices mentions `annotationlib.get_annotations()` as superseding the inspect module function, but this is only introduced in python 3.14. 

This can be seen by running `python/test/unit/language/test_frontend.py`, however interestingly `python/test/unit/language/test_annotations.py` passes fine. 

```
============================================= short test summary info ==============================================
FAILED python/test/unit/language/test_frontend.py::test_retrieve_item - triton.compiler.errors.CompilationError: at 6:11:
FAILED python/test/unit/language/test_frontend.py::test_jit_method - triton.compiler.errors.CompilationError: at 7:11:
FAILED python/test/unit/language/test_frontend.py::test_jit_getitem - triton.compiler.errors.CompilationError: at 4:8:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_initializers - triton.compiler.errors.CompilationError: at 3:12:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_auto_init_assigns_members - TypeError: State.__init__() takes 1 positional arguments but 2 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_auto_init_with_tuples - TypeError: State.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_auto_init_respects_user_defined_init - AttributeError: State has no attribute 'x'
FAILED python/test/unit/language/test_frontend.py::test_aggregate_with_constexpr - triton.compiler.errors.CompilationError: at 4:10:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_with_tuple - triton.compiler.errors.CompilationError: at 4:10:
FAILED python/test/unit/language/test_frontend.py::test_late_bound_class_reference - triton.compiler.errors.CompilationError: at 4:12:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_constexpr_function - triton.compiler.errors.CompilationError: at 2:10:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_field_inheritance - TypeError: Child.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_multilevel_inheritance - TypeError: Child.__init__() takes 1 positional arguments but 4 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_field_inheritance_with_methods - TypeError: Child.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_default_values - TypeError: WithDefaults.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_replace - TypeError: State.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_replace_invalid_field - TypeError: State.__init__() takes 1 positional arguments but 2 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_inherited_defaults - TypeError: Child.__init__() got an unexpected keyword argument 'y'
FAILED python/test/unit/language/test_frontend.py::test_aggregate_string_annotations_resolved - TypeError: StringAnnoChild.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_default_value_auto_wrapped - TypeError: State.__init__() takes 1 positional arguments but 2 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_post_construction_immutable - TypeError: State.__init__() takes 1 positional arguments but 3 were given
FAILED python/test/unit/language/test_frontend.py::test_aggregate_inheritance_ir - triton.compiler.errors.CompilationError: at 7:12:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_inherited_method_ir - triton.compiler.errors.CompilationError: at 7:12:
FAILED python/test/unit/language/test_frontend.py::test_aggregate_replace_ir - triton.compiler.errors.CompilationError: at 7:12:
========================================== 24 failed, 35 passed in 2.27s ===========================================
```



### Environment details

Triton 3.7.0 trunk as of commit `1859e68`
GPU: N/A

--- a/python/triton/language/core.py
+++ b/python/triton/language/core.py
@@ -1609,7 +1609,6 @@ def _resolve_aggregate_fields(cls):
 
     # Add cls's own fields, resolving string annotations via typing.get_type_hints.
     own_names = inspect.get_annotations(cls)
-
     hints = typing.get_type_hints(cls)
     for name in own_names:
         all_annotations[name] = hints[name]

--- a/python/triton/language/core.py
+++ b/python/triton/language/core.py
@@ -1604,11 +1604,12 @@ def _resolve_aggregate_fields(cls):
             continue
         if not getattr(base, "__triton_aggregate__", False):
             raise TypeError(f"Aggregates can only inherit from other aggregates, but got non-aggregate base: {base}")
-        all_annotations.update(getattr(base, "__annotations__", {}))
+        all_annotations.update(inspect.get_annotations(base))
         all_defaults.update(getattr(base, "__aggregate_defaults__", {}))
 
     # Add cls's own fields, resolving string annotations via typing.get_type_hints.
-    own_names = cls.__dict__.get("__annotations__", {})
+    own_names = inspect.get_annotations(cls)
+
     hints = typing.get_type_hints(cls)
     for name in own_names:
         all_annotations[name] = hints[name]

--- a/python/triton/runtime/interpreter.py
+++ b/python/triton/runtime/interpreter.py
@@ -1311,7 +1311,7 @@ class GridExecutor:
         self.arg_names = arg_names
         self.grid = grid
         self.pre_run_hooks = pre_run_hooks
-        __annotations__ = {name: _normalize_ty(ty) for name, ty in fn.__annotations__.items()}
+        __annotations__ = {name: _normalize_ty(ty) for name, ty in inspect.get_annotations(fn).items()}
         self.constexprs = [name for name in arg_names if __annotations__.get(name) == "constexpr"]
 
     def _init_args_hst(self, args_dev, kwargs):

